src/main/kotlin/lotto/model/LottoMachine.kt

```diff
@@ -0,0 +1,122 @@
+package lotto.model
+
+import lotto.service.AmountValidator
+import lotto.service.Constants
+import lotto.view.InputView
+import lotto.view.OutputView
+import lotto.service.Rank
+
+object LottoMachine {
+    fun start() {
+        val amount = getAmountInput()
+        val manualTicketCount = getManualTicketCount(amount)
+        val manualTickets = getManualTickets(manualTicketCount)
+        val automaticTickets = generateAutomaticTickets(amount, manualTicketCount)
+        val winningNumbers = InputView.inputWinningNumbers()
+        val bonusNumber = InputView.inputBonusNumber(winningNumbers)
+        calculateStats(automaticTickets + manualTickets, winningNumbers, bonusNumber)
+        OutputView.displayResults(winStats, calculateReturnRate(amount))
+    }
+
+    private fun getAmountInput(): Int {
+        while (true) {
+            try {
+                val amount = InputView.inputPurchaseAmount()
+                AmountValidator.validate(amount)
+                return amount
+            } catch (_: IllegalArgumentException) {
+                println("Enter a valid amount to buy tickets")
+            }
+        }
+    }
+
+    private fun getManualTicketCount(amount: Int): Int {
+        while (true) {
+            try {
+                val manualTicketCount = InputView.inputManualTicketCount()
+                ManualTicket.checkTicketCount(manualTicketCount, amount)
+                return manualTicketCount
+            } catch (_: IllegalArgumentException) {
+                println("Enter a valid manual ticket count")
+            }
+        }
+    }
+
+    private fun getManualTickets(manualTicketCount: Int): List<Ticket> {
+        while (true) {
+            try {
+                val manualTicketsString = InputView.inputManualTicketNumbers(manualTicketCount)
+                val manualTickets = ManualTicket.generateManualTickets(manualTicketsString)
+                return manualTickets
+            } catch (_: IllegalArgumentException) {
+                println("Enter valid manual tickets")
+            }
+        }
+    }
+
+    private fun generateAutomaticTickets(
+        amount: Int,
+        manualTicketCount: Int,
+    ): List<Ticket> {
+        val ticketCount = amount / Constants.TICKET_PRICE - manualTicketCount
+        val ticketsList = mutableListOf<Ticket>()
+        repeat(ticketCount) {
+            val ticket = Ticket(generateTicketNumbers())
+            ticketsList.add(ticket)
+        }
+        OutputView.displayTickets(ticketCount, ticketsList, manualTicketCount)
+        return ticketsList
+    }
+
+    private fun generateTicketNumbers(): List<LottoNumber> {
+        return LottoNumber.Companion.takeRandom(Constants.NUMBER_COUNT)
```

Comment:
```suggestion
        return LottoNumber.takeRandom(Constants.NUMBER_COUNT)
```

`Companion` keyword isn't necessary when we call methods or use the variables in it.

Reply:
Thank you for catching that! Is Companion object like static so Members inside the companion object can be accessed using the class name?

Reply:
Yes, we can access them using the class name,
and kinda-yes about it being `like static`. (This is where `@JvmStatic` comes in)
I recommend [checking this out](https://kotlinlang.org/docs/object-declarations.html#companion-objects). It explains what companion object is in detail, with easy-to-read examples 👍